frontend/src/App.tsx

```diff
@@ -29,22 +28,28 @@ export default function App() {
         axios.delete("/api/books/" + id)
             .then((response) => response.status === 200 && fetchBooks())
             .catch((error) => console.log(error.message))
-
-
     }
 
     useEffect(() => {
         fetchBooks()
     }, []);
 
+    const [searchInput, setSearchInput] = useState("")
+
+    const filteredBooks: Book[] = data
+        .filter((book) => book.title.toLowerCase().includes(searchInput.toLowerCase()) ||
+            book.author.toLowerCase().includes(searchInput.toLowerCase()));
+
```

Comment:
To make sure if the data is available when the app is running, I suggest the following code.

```typescript
const filteredBooks: Book[] = data.length > 0 ?
        data.filter((book) => book.title?.toLowerCase().includes(searchInput.toLowerCase()) ||
            book.author?.toLowerCase().includes(searchInput.toLowerCase()))
        : [];
``` `

Reply:
but wouldn't it be ok if data is just an empty array? then there is just nothing to filter